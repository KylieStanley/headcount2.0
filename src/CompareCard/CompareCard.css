.compare {
  height: 198px;
  text-align: center;
  display: flex;
  justify-content: center;
  margin: 73px 8px 78px;
}