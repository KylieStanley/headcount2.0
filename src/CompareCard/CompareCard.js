import React from 'react';
import './CompareCard.css'


const CompareCard = ({ selectedCards, comparedDistricts, district1, district2 }) => {
 
  if (selectedCards.length === 2) {
    var compared = 
      <div className="card compare">
        <h3>{ district1 } : { comparedDistricts[district1] } </h3>
        <h3>&lt;--- { comparedDistricts.compared } ---></h3>
        <h3>{ district2 } : { comparedDistricts[district2] } </h3>
      </div> 
  }

   const waitingToCompare = <div className="card compare">Choose Two Districts to Compare!</div>

  return (
    compared ? compared : waitingToCompare 
  )
}

export default CompareCard;
