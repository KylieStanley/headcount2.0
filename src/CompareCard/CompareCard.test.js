import React from 'react';
import CompareCard from './CompareCard.js';
import {shallow, mount} from 'enzyme';

describe('CompareCard', () => {
  it('should match the snapshot with data passed in correctly', () => {
    const wrapper = shallow( <CompareCard 
      selectedCards={[{location: "COLORADO SPRINGS",
                       stats: {2005: .4}
                      },
                      {location: "COLORADO",
                       stats: {2006: .4}
                      }]} 
      district1='COLORADO' 
      district2='COLORADO SPRINGS'
      comparedDistricts={1} 
      key={3}
    />)

    expect(wrapper).toMatchSnapshot();
  })
})