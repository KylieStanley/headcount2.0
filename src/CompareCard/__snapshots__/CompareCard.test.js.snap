// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompareCard should match the snapshot with data passed in correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CompareCard
    comparedDistricts={1}
    district1="COLORADO"
    district2="COLORADO SPRINGS"
    selectedCards={
        Array [
            Object {
              "location": "COLORADO SPRINGS",
              "stats": Object {
                "2005": 0.4,
              },
            },
            Object {
              "location": "COLORADO",
              "stats": Object {
                "2006": 0.4,
              },
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          COLORADO
           : 
           
</h3>,
        <h3>
          &lt;--- 
           ---&gt;
</h3>,
        <h3>
          COLORADO SPRINGS
           : 
           
</h3>,
      ],
      "className": "card compare",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "COLORADO",
            " : ",
            undefined,
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          "COLORADO",
          " : ",
          undefined,
          " ",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "<--- ",
            undefined,
            " --->",
          ],
        },
        "ref": null,
        "rendered": Array [
          "<--- ",
          undefined,
          " --->",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "COLORADO SPRINGS",
            " : ",
            undefined,
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          "COLORADO SPRINGS",
          " : ",
          undefined,
          " ",
        ],
        "type": "h3",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            COLORADO
             : 
             
</h3>,
          <h3>
            &lt;--- 
             ---&gt;
</h3>,
          <h3>
            COLORADO SPRINGS
             : 
             
</h3>,
        ],
        "className": "card compare",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "COLORADO",
              " : ",
              undefined,
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            "COLORADO",
            " : ",
            undefined,
            " ",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "<--- ",
              undefined,
              " --->",
            ],
          },
          "ref": null,
          "rendered": Array [
            "<--- ",
            undefined,
            " --->",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "COLORADO SPRINGS",
              " : ",
              undefined,
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            "COLORADO SPRINGS",
            " : ",
            undefined,
            " ",
          ],
          "type": "h3",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
